fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+
+For instance, for VISIONPRESCRIPTION there are: 
+
+```
+VISIONPRESCRIPTION_NUMBER_VALUES
+VISIONPRESCRIPTION_DATE_VALUES
+VISIONPRESCRIPTION_STR_VALUES
+VISIONPRESCRIPTION_TOKEN_VALUES
+VISIONPRESCRIPTION_COMPOSITES
+VISIONPRESCRIPTION_QUANTITY_VALUES
+VISIONPRESCRIPTION_LATLNG_VALUES
+```
+
+There are also additional tables to support search: `LOGICAL_RESOURCE`, `RESOURCE_TYPES` and `PARAMETER_NAMES` These tables are paritioned, and are specific for each tenant. 
+
+Each of these tables follows a common pattern, and are defined in the `FHIRResourceTableGroup.java`. 
+
+For any changes to any tables or indices, the table definition must be updated to a more recent version, one must update the table definition. For instance, for `addStrValues` one must increment the `setVersion` number: 
+
+``` java
+// Parameters are tied to the logical resource
+Table tbl = Table.builder(schemaName, tableName)
+    .setVersion(2)
+    .addTag(FhirSchemaTags.RESOURCE_TYPE, prefix)
+    .setTenantColumnName(MT_ID)
+    .addBigIntColumn(             ROW_ID,      false)
+    .addIntColumn(     PARAMETER_NAME_ID,      false)
+    .addVarcharColumn(         STR_VALUE, msb,  true)
+    .addVarcharColumn(   STR_VALUE_LCASE, msb,  true)
+    .addBigIntColumn(LOGICAL_RESOURCE_ID,      false)
+    .addIndex(IDX + tableName + "_PSR", PARAMETER_NAME_ID, STR_VALUE, LOGICAL_RESOURCE_ID)
+    .addIndex(IDX + tableName + "_PLR", PARAMETER_NAME_ID, STR_VALUE_LCASE, LOGICAL_RESOURCE_ID)
+    .addIndex(IDX + tableName + "_RPS", LOGICAL_RESOURCE_ID, PARAMETER_NAME_ID, STR_VALUE)
+    .addIndex(IDX + tableName + "_RPL", LOGICAL_RESOURCE_ID, PARAMETER_NAME_ID, STR_VALUE_LCASE)
+    .addPrimaryKey(PK + tableName, ROW_ID)
+    .setIdentityColumn(ROW_ID, Generated.BY_DEFAULT)
+    .addForeignKeyConstraint(FK + tableName + "_PNID", schemaName, PARAMETER_NAMES, PARAMETER_NAME_ID)
+    .addForeignKeyConstraint(FK + tableName + "_RID", schemaName, logicalResourcesTable, LOGICAL_RESOURCE_ID)
+    .setTablespace(fhirTablespace)
+    .addPrivileges(resourceTablePrivileges)
+    .enableAccessControl(this.sessionVariable)
+    .build(model);
+```
+
+If a new resource is added to the specification, the schema utility automatically provisions it on the next execution of the update schema actions. 
+
+Common reasons to modify the Resource's tables are: 
+- **Specification Change (Version-to-Version changes)**
+    - The Resources are saved as Blobs in the database, and are transparent to the FHIR Version changes. The changes from version-to-version are resilient to field add-remove changes, type changes, and resources additions.  This migration must be done manually.
+    - Resource removals must be done manually. 
+- **Column Attribute Changes (Space, Value Type)**
+    - As the column types, changes are changed in any column, a specific alter statement must be executed on the table, and constraints must be changed or relaxed. This migration must be done manually.
```

Comment:
> As the column types, changes are changed in any column
I didn't get what this was trying to say

Reply:
I revised